src/Tokenizer/Tokens.php

```diff
@@ -300,14 +300,7 @@ public static function getBlockEdgeDefinitions(): array
     #[\ReturnTypeWillChange]
     public function setSize($size): bool
     {
-        if (\count($this) !== $size) {
-            $this->changed = true;
-            $this->namespaceDeclarations = null;
-
-            return parent::setSize($size);
-        }
-
-        return true;
+        throw new \RuntimeException(\sprintf('Do not change tokens collection size using %s.', __METHOD__));
```

Comment:
small nit: I prefer exception message like: `Changing tokens collection size explicitly is not allowed.` + it should be tested